tests/unit/test_blob.py

```diff
@@ -1906,10 +1911,14 @@ def _do_multipart_success(
 
         upload_url += "?" + urlencode(qs_params)
 
+        blob_data = b'{"name": "blob-name"}\r\n'
+        if metadata:
+            blob_data = b'{"name": "blob-name", "metadata": {"test": "test"}}\r\n'
```

Comment:
This needs to use the passed-in `metadata` argument, e.g.:
```suggestion
            blob_data = b'{"name": "blob-name", "metadata": %r}\r\n' % (metadata,)
```